packages/dm-core-plugins/src/view_selector/useViewSelector.tsx

```diff
@@ -52,19 +52,25 @@ export function useViewSelector(
     onSubmitAdded?: (data: any) => void,
     onChangeAdded?: (data: any) => void
   ) => {
-    if (!viewSelectorItems.find((view: TItemData) => view.viewId === viewId)) {
+    const existingViewIndex = viewSelectorItems.findIndex(
+      (view: TItemData) => view.viewId === viewId
+    )
+    const newView: TItemData = {
+      viewId: viewId,
+      viewConfig: viewConfig,
+      label: viewConfig.label ?? viewId,
+      rootEntityId: rootId || idReference,
+      onSubmit: onSubmitAdded,
+      onChange: onChangeAdded,
+      closeable: true,
+      isSubItem: isSubItem,
+    }
+    if (existingViewIndex === -1) {
       // View does not exist, add it
-      const newView: TItemData = {
-        viewId: viewId,
-        viewConfig: viewConfig,
-        label: viewConfig.label ?? viewId,
-        rootEntityId: rootId || idReference,
-        onSubmit: onSubmitAdded,
-        onChange: onChangeAdded,
-        closeable: true,
-        isSubItem: isSubItem,
-      }
       setViewSelectorItems([...viewSelectorItems, newView])
+    } else {
+      newView.viewId = viewSelectorItems[existingViewIndex].viewId
+      viewSelectorItems[existingViewIndex] = newView
```

Comment:
How can this work, when you are not calling "setViewSelectorItems()"?

Reply:
It works, but mutating state is not recommended. Setting state again is better 👍